rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceSpec.scala

```diff
@@ -1014,6 +1014,41 @@ class ReduceSpec extends FlatSpec with Matchers with PersistentStoreTester {
     errorLog.readAndClearErrorVector should be(Vector.empty[InterpreterError])
   }
 
+  it should "substitute before serialization" in {
+    implicit val errorLog = new ErrorLog()
+    implicit val costAccounting =
+      CostAccountingAlg.unsafe[Task](CostAccount.zero)
+    val splitRand = rand.splitByte(0)
+    val unsubProc: Par =
+      New(bindCount = 1, p = EVar(BoundVar(1)), locallyFree = BitSet(0))
+    val subProc: Par =
+      New(bindCount = 1, p = GPrivate("zero"), locallyFree = BitSet())
+    val serializedProcess         = subProc.toByteString
+    val toByteArrayCall: Par      = EMethod("toByteArray", unsubProc, List[Par]())
+    def wrapWithSend(p: Par): Par = Send(Quote(GString("result")), List[Par](p), false, BitSet())
```

Comment:
isn't `locallyFree` set a union of `channel`'s `locallyFree` and `locallyFree` of the body?

Reply:
Good catch.